src/ui/QuerySuggestPreview/QuerySuggestPreview.ts

```diff
@@ -1,59 +1,272 @@
-    numberOfPreviewResults: ComponentOptions.buildNumberOption({ defaultValue: 0, min: 0 })
+    resultTemplate: TemplateComponentOptions.buildTemplateOption(),
+    /**
+     * The maximum number of query results to render in the preview.
+     *
+     * **Minimum value:** `1`
+     * **Maximum value:** `6`
+     * **Default value:** `3`
+     */
+    numberOfPreviewResults: ComponentOptions.buildNumberOption({
+      defaultValue: 3,
+      min: 1,
+      max: 6
+    }),
+    /**
+     * Width of the preview container (in pixels).
+     *
+     * If this option is `undefined` or lower than the remaning space left by the suggestion,
+     * the component takes all the space left over by the query suggestions.
+     */
+    previewWidth: ComponentOptions.buildNumberOption(),
+    /**
+     *  Width of the suggestion container (in pixels or percentage).
+     *
+     * **Default value:** `33%`
+     */
+    suggestionWidth: ComponentOptions.buildStringOption({ defaultValue: '33%' }),
+    /**
+     *  The text to display at the top of the preview, which is followed by "`<SUGGESTION>`", where `<SUGGESTION>` is the hovered query suggestion.
+     *
+     * **Default value:** `Query result items for`
+     */
+    headerText: ComponentOptions.buildLocalizedStringOption({ defaultValue: 'QuerySuggestPreview' }),
+    /**
+     *  The hovering time (in ms) required on a query suggestion before requesting preview items.
+     *
+     * **Default value:** `200`
+     */
+    executeQueryDelay: ComponentOptions.buildNumberOption({ defaultValue: 200 })
   };
 
-  constructor(public element: HTMLElement, public options?: IQuerySuggestPreview, public bindings?: IComponentBindings) {
-    super(element, options, bindings, QuerySuggestPreview.ID);
+  private previousSuggestionHovered: string;
+  private renderer: ResultListRenderer;
+  private timer;
+  public currentlyDisplayedResults: IQueryResult[] = [];
 
-    //Have to do this for now or or break the search interface
-    //Will fix in JSUI-2508 where i will use my template
-    this.element.style.display = 'none';
+  /**
+   * Creates a new QuerySuggestPreview component.
+   * @param element The HTMLElement on which to instantiate the component.
+   * @param options The options for the QuerySuggestPreview component.
+   * @param bindings The bindings that the component requires to function normally. If not set, these will be
+   * automatically resolved (with a slower execution time).
+   */
+  constructor(public element: HTMLElement, public options?: IQuerySuggestPreview, public bindings?: IComponentBindings) {
+    super(element, QuerySuggestPreview.ID, bindings);
 
     this.options = ComponentOptions.initComponentOptions(element, QuerySuggestPreview, options);
 
+    if (!this.options.resultTemplate) {
+      this.logger.warn(
+        `Option "resultTemplate" is *REQUIRED* on the component "QuerySuggestPreview". The component or the search page might *NOT WORK PROPERLY*`,
```

Comment:
Suggestion:

> Specifying a result template is required for the 'QuerySuggestPreview' component to work properly. See [Providing Query Suggestion Result Previews](https://docs.coveo.com/340/#providing-query-suggestion-result-previews).

Reply:
I can use that no problem but the link is dead, is this something you will create with that URL later?

Reply:
Oops the link was wrong!

should be:

https://docs.coveo.com/en/340/#providing-query-suggestion-result-previews

The anchor won't work for now (I'll add the subsection later), but at least you'll get to an existing page :)

Reply:
I also suggested the same link in another comment; can you update the URL there as well?

Thanks!

Reply:
will do